.github/workflows/markdown-checks.yaml

```diff
@@ -0,0 +1,47 @@
+name: Markdown linter
+
+# What the execution context is:
+# The creation of a pull request. Can also be triggered manually.
+
+# What it does:
+# Performs Markdown linting on the content of the pull request.
+
+# Why we need it:
+# To maintain a level of consistency in the markup content.
+
+# What's important to know:
+# As well as the rules defined in the configuration, a custom
+# rule is also employed to check title case.
+
+on:
+  pull_request:
+    branches: [main]
+
+jobs:
+  main:
+    runs-on: ubuntu-20.04
+    steps:
+
+    - id: checkout_repo
+      name: Check out the repository content
+      uses: actions/checkout@v2
+
+    - id: add_matcher
+      name: Add the matcher for markdownlint style message output
+      run: "echo ::add-matcher::.github/workflows/markdownlint/problem-matcher.json"
+
+    - id: install_linter
+      name: Install linting tool and custom rule
+      run: |
+        npm install \
+          --no-package-lock \
+          --no-save \
+          markdownlint-cli markdownlint-rule-titlecase
+
+    - id: run_linter
+      name: Run linter with specific rules, on the docs/ content
+      run: |
+        npx markdownlint \
+          --config .github/workflows/markdownlint/config.yaml \
+          --rules markdownlint-rule-titlecase \
+          docs/
```

Comment:
An alternative would be to add it a proper package file and include them in there. The advantage would be that we could pin the versions and don't always use the latest. (Especially ensuring that a change in output format doesn't break the problem-matcher)
(or even go as far and creating and SAP-docs/markdownlinter action, which bundles everything for easier and consistent reuse across all doc repos (installing markdownlint/custom rules, setup problem matcher, execute scan).

Reply:
This is a nice idea; I think we should consider this for the next phase - in particular the SAP-own action.